test/tracer.js

```diff
@@ -249,4 +252,47 @@ describe('tracer should', () => {
             done();
         });
     });
+
+    describe('Metrics', () => {
+        it ('startInternalSpan', () => {
+            let params = [
+                { 'hasParent': false, 'context': '1:2:0:1', 'sampled': true, 'metrics': ['spansStarted', 'spansSampled', 'tracesStartedSampled']},
+                { 'hasParent': true, 'context': '1:2:100:1', 'sampled': true, 'metrics': ['spansStarted', 'spansSampled', 'tracesJoinedSampled']},
+                { 'hasParent': false, 'context': '1:2:0:0', 'sampled': false, 'metrics': ['spansStarted', 'spansNotSampled', 'tracesStartedNotSampled']},
+                { 'hasParent': true, 'context': '1:2:100:0', 'sampled': false, 'metrics': ['spansStarted', 'spansNotSampled', 'tracesJoinedNotSampled']},
+            ];
```

Comment:
this is the problem with writing unit tests against internal methods - you fake the arguments to satisfy the unit test, but you're not checking what really happens when the public API is used. Can you augment this test to actually use the public `startSpan()`?